### PR TITLE
Rework cartridge template

### LINKED ISSUE #428: [2pt] Make Cartridge application template great again
I think it's time to review and rework default application code.

See also #296, #300

Note about https://github.com/tarantool/cartridge/issues/1189, specify the same ports in `instances.yml`.

### PR DESCRIPTION
**Pull request by mRrvz**
Reworked ``cartridge create`` template

* Registration ``admin`` function and path configuration moved to separated files
* Update and improve ``helper`` in tests
* Changed the standard port of the ``stateboard``
* And other cosmetic fixes

Closes #428

### CODE REVIEW

File: cli/create/templates/cartridge/deps.sh [hunk 1]
```diff
@@ -7,3 +7,4 @@ set -e
 tarantoolctl rocks install luatest 0.5.0
 tarantoolctl rocks install luacov 0.13.0
 tarantoolctl rocks install luacheck 0.25.0
+tarantoolctl rocks install prometheus 1.0.4
```
Comment (dokshina) [inline cli/create/templates/cartridge/deps.sh:10]: It's already placed in rockspec.

File: cli/create/templates/cartridge/{{ .Name }}-scm-1.rockspec [hunk 2]
```diff
@@ -11,6 +11,7 @@ dependencies = {
     'cartridge == 2.4.0-1',
     'metrics == 0.6.1-1',
     'cartridge-cli-extensions == 1.1.0-1',
+    'prometheus == 1.0.4',
```
Comment (olegrok) [inline cli/create/templates/cartridge/{{ .Name }}-scm-1.rockspec:14]: Please, note that this module is deprecated (https://github.com/tarantool/prometheus#deprecated)

Use prometheus plugin from metrics.

File: cli/create/templates/cartridge/app/admin.lua [hunk 3]
```diff
@@ -0,0 +1,41 @@
+-- register admin function probe to use it with "cartridge admin"
+
+local function register_admin()
+    local cli_admin = require('cartridge-cli-extensions.admin')
```
Comment (dokshina) [inline cli/create/templates/cartridge/app/admin.lua:4]: Move require to the top of file.

File: cli/create/templates/cartridge/app/path.lua [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        package.cpath = app_dir .. '/?.dylib;' .. package.cpath
+        package.cpath = app_dir .. '/.rocks/lib/tarantool/?.so;' .. package.cpath
+        package.cpath = app_dir .. '/.rocks/lib/tarantool/?.dylib;' .. package.cpath
+    end
+
+    return { path = package.path, cpath = package.cpath, root = package.searchroot()}
+end
+
+local path = {cfg = configure_path}
+return path
```
Comment (dokshina) [inline cli/create/templates/cartridge/app/path.lua:30]: Please, add a newline.

Comment (dokshina) [inline cli/create/templates/cartridge/app/path.lua:30]: `return {
   cfg = cfg, -- I would like this function to be named cfg
}`

File: cli/create/templates/cartridge/init.lua [hunk 5]
```diff
@@ -2,29 +2,14 @@
...
-    package.path = app_dir .. '/?.lua;' .. package.path
-    package.path = app_dir .. '/?/init.lua;' .. package.path
-    package.path = app_dir .. '/.rocks/share/tarantool/?.lua;' .. package.path
-    package.path = app_dir .. '/.rocks/share/tarantool/?/init.lua;' .. package.path
-    package.cpath = app_dir .. '/?.so;' .. package.cpath
-    package.cpath = app_dir .. '/?.dylib;' .. package.cpath
-    package.cpath = app_dir .. '/.rocks/lib/tarantool/?.so;' .. package.cpath
-    package.cpath = app_dir .. '/.rocks/lib/tarantool/?.dylib;' .. package.cpath
-end
+-- configure path
```
Comment (dokshina) [inline cli/create/templates/cartridge/init.lua:5]: I don't understand what the path is configured.
As I remember, this code is required to run application from other directories (e.g. when you start it after deploying RPM), please describe it in your comment.

File: cli/create/templates/cartridge/init.lua [hunk 6]
```diff
@@ -2,29 +2,14 @@
...
+
+local path = require('app.path')
+local config = path.cfg()
+
+package.path = config.path
+package.cpath = config.cpath
+package.setsearchroot(config.root)
```
Comment (dokshina) [inline cli/create/templates/cartridge/init.lua:12]: Why do you do it here? We've moved this code to make `init.lua` more clear and simple. I expect smth like.
 
```lua
-- configure setsearchroot
require('app.path').cfg()
```

File: cli/create/templates/cartridge/init.lua [hunk 7]
```diff
@@ -42,39 +27,7 @@ local ok, err = cartridge.cfg({
...
-            string.format('Probe %q: OK', opts.uri),
-        }
-    end,
-}
+-- register admin function to use it with 'cartridge admin' command
 
-local ok, err = cli_admin.register('probe', probe.usage, probe.args, probe.call)
-assert(ok, err)
+local admin = require('app.admin')
+admin.register()
```
Comment (dokshina) [inline cli/create/templates/cartridge/init.lua:33]: Maybe `admin.init()`? In fact, we initialize admin functions here, registering it in `cli_admin` module is just a part of initialization.

File: CHANGELOG.md [hunk 8]
```diff
@@ -15,6 +15,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Changed
 
 - Update `metrics` to `0.7.0`
+- Reworked ``cartridge create`` template:
+  * Registration ``admin`` function and path configuration moved to separated files
+  * Update and better ``helper`` in tests
```
Comment (olegrok) [inline CHANGELOG.md:20]: `Update and improve` may be?

verb and adjective looks bad.

File: cli/create/templates/cartridge/app/admin.lua [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+            }
+        end,
+    }
+
+    local ok, err = cli_admin.register('probe', probe.usage, probe.args, probe.call)
+    assert(ok, err)
+end
+
+local admin = {init = init}
+return admin
```
Comment (olegrok) [inline cli/create/templates/cartridge/app/admin.lua:40]: why not `return {init = init}`?

File: cli/create/templates/cartridge/app/path.lua [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        package.path = app_dir .. '/.rocks/share/tarantool/?/init.lua;' .. package.path
+        package.cpath = app_dir .. '/?.so;' .. package.cpath
+        package.cpath = app_dir .. '/?.dylib;' .. package.cpath
+        package.cpath = app_dir .. '/.rocks/lib/tarantool/?.so;' .. package.cpath
+        package.cpath = app_dir .. '/.rocks/lib/tarantool/?.dylib;' .. package.cpath
+    end
+end
+
+local path = {cfg = cfg}
+return path
```
Comment (olegrok) [inline cli/create/templates/cartridge/app/path.lua:31]: Why not `return {cfg = cfg}`?

File: README.rst [hunk 11]
```diff
@@ -989,7 +989,7 @@ Add the ``myapp-stateboard`` section to ``/etc/tarantool/conf.d/myapp.yml``:
 .. code-block:: yaml
 
     myapp-stateboard:
-      listen: localhost:3310
+      listen: localhost:4401
```
Comment (olegrok) [inline README.rst:992]: Is there a motivation for this change?

File: cli/commands/start.go [hunk 12]
```diff
@@ -57,7 +57,7 @@ func init() {
 func runStartCmd(cmd *cobra.Command, args []string) error {
 	var err error
 
-	if timeoutStr == "" && !ctx.Running.Daemonize {
+	if timeoutStr != "" && !ctx.Running.Daemonize {
```
Comment (olegrok) [inline cli/commands/start.go:60]: It's a behaviour change. But seems it's not tested anyhow and not described in changelog...
Also looks like it's not related to your current patch and should be pushed separately.

**Review by olegrok — COMMENTED** [review]
Thanks for fixes. Couple of questions below.

File: README.rst (hunk 11 above)
Comment (mRrvz) [reply README.rst]: Removed

**Review by olegrok — APPROVED** [review]

File: cli/commands/start.go (hunk 12 above)
Comment (mRrvz) [reply cli/commands/start.go]: https://github.com/tarantool/cartridge-cli/pull/471
